web/src/app/services/ServiceDeleteDialog.js

```diff
@@ -61,34 +63,42 @@ export default function ServiceDeleteDialog({ serviceID, onClose }) {
     variables: { id: serviceID },
   })
   const input = [{ type: 'service', id: serviceID }]
-  const dispatch = useDispatch()
   const [deleteService, { loading, error }] = useMutation(mutation, {
     variables: { input },
-    onCompleted: () => dispatch(push('/services')),
   })
 
-  if (dataLoading) return <Spinner />
+  const epID = _.get(data, 'service.ep.id')
+  const epName = _.get(
+    data,
+    'service.ep.name',
+    <Spinner text='fetching policy...' />,
+  )
 
   if (deleteEP) {
     input.push({
       type: 'escalationPolicy',
-      id: data.service.escalationPolicyID,
+      id: epID,
     })
   }
 
   return (
     <FormDialog
       title='Are you sure?'
       confirm
-      subTitle={`This will delete the service: ${data.service.name}`}
+      subTitle={
```

Comment:
This looks to be throwing a warning if the spinner is returned: ```Failed prop type: Invalid prop `subTitle` of type `object` supplied to `FormDialog`, expected `string`.```. So we should update that to a `p.node` propType.

May have to make further adjustments though as this bubbles down to `DialogTitleWrapper` and `subTitle` is wrapped by `Typography`:
```
<DialogContent className={classes.subtitle}>
  <Typography variant='subtitle1' component='p'>
    {subTitle}
  </Typography>
</DialogContent>
```